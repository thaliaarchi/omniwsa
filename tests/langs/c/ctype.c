#include <ctype.h>
#include <limits.h>
#include <stdio.h>

#define CHECK(c, is) if (is(c)) { printf(" " #is); }

int main() {
    for (int c = 0; c <= UCHAR_MAX; c++) {
        printf("0x%02X", c);
        CHECK(c, isalnum);
        CHECK(c, isalpha);
        CHECK(c, isblank);
        CHECK(c, iscntrl);
        CHECK(c, isdigit);
        CHECK(c, isgraph);
        CHECK(c, islower);
        CHECK(c, isprint);
        CHECK(c, ispunct);
        CHECK(c, isspace);
        CHECK(c, isupper);
        CHECK(c, isxdigit);
        printf("\n");
    }
    return 0;
}
