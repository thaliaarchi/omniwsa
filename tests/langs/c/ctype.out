Apple clang version 16.0.0 (clang-1600.0.26.4)
0x00 iscntrl
0x01 iscntrl
0x02 iscntrl
0x03 iscntrl
0x04 iscntrl
0x05 iscntrl
0x06 iscntrl
0x07 iscntrl
0x08 iscntrl
0x09 isblank iscntrl isspace
0x0A iscntrl isspace
0x0B iscntrl isspace
0x0C iscntrl isspace
0x0D iscntrl isspace
0x0E iscntrl
0x0F iscntrl
0x10 iscntrl
0x11 iscntrl
0x12 iscntrl
0x13 iscntrl
0x14 iscntrl
0x15 iscntrl
0x16 iscntrl
0x17 iscntrl
0x18 iscntrl
0x19 iscntrl
0x1A iscntrl
0x1B iscntrl
0x1C iscntrl
0x1D iscntrl
0x1E iscntrl
0x1F iscntrl
0x20 isblank isprint isspace
0x21 isgraph isprint ispunct
0x22 isgraph isprint ispunct
0x23 isgraph isprint ispunct
0x24 isgraph isprint ispunct
0x25 isgraph isprint ispunct
0x26 isgraph isprint ispunct
0x27 isgraph isprint ispunct
0x28 isgraph isprint ispunct
0x29 isgraph isprint ispunct
0x2A isgraph isprint ispunct
0x2B isgraph isprint ispunct
0x2C isgraph isprint ispunct
0x2D isgraph isprint ispunct
0x2E isgraph isprint ispunct
0x2F isgraph isprint ispunct
0x30 isalnum isdigit isgraph isprint isxdigit
0x31 isalnum isdigit isgraph isprint isxdigit
0x32 isalnum isdigit isgraph isprint isxdigit
0x33 isalnum isdigit isgraph isprint isxdigit
0x34 isalnum isdigit isgraph isprint isxdigit
0x35 isalnum isdigit isgraph isprint isxdigit
0x36 isalnum isdigit isgraph isprint isxdigit
0x37 isalnum isdigit isgraph isprint isxdigit
0x38 isalnum isdigit isgraph isprint isxdigit
0x39 isalnum isdigit isgraph isprint isxdigit
0x3A isgraph isprint ispunct
0x3B isgraph isprint ispunct
0x3C isgraph isprint ispunct
0x3D isgraph isprint ispunct
0x3E isgraph isprint ispunct
0x3F isgraph isprint ispunct
0x40 isgraph isprint ispunct
0x41 isalnum isalpha isgraph isprint isupper isxdigit
0x42 isalnum isalpha isgraph isprint isupper isxdigit
0x43 isalnum isalpha isgraph isprint isupper isxdigit
0x44 isalnum isalpha isgraph isprint isupper isxdigit
0x45 isalnum isalpha isgraph isprint isupper isxdigit
0x46 isalnum isalpha isgraph isprint isupper isxdigit
0x47 isalnum isalpha isgraph isprint isupper
0x48 isalnum isalpha isgraph isprint isupper
0x49 isalnum isalpha isgraph isprint isupper
0x4A isalnum isalpha isgraph isprint isupper
0x4B isalnum isalpha isgraph isprint isupper
0x4C isalnum isalpha isgraph isprint isupper
0x4D isalnum isalpha isgraph isprint isupper
0x4E isalnum isalpha isgraph isprint isupper
0x4F isalnum isalpha isgraph isprint isupper
0x50 isalnum isalpha isgraph isprint isupper
0x51 isalnum isalpha isgraph isprint isupper
0x52 isalnum isalpha isgraph isprint isupper
0x53 isalnum isalpha isgraph isprint isupper
0x54 isalnum isalpha isgraph isprint isupper
0x55 isalnum isalpha isgraph isprint isupper
0x56 isalnum isalpha isgraph isprint isupper
0x57 isalnum isalpha isgraph isprint isupper
0x58 isalnum isalpha isgraph isprint isupper
0x59 isalnum isalpha isgraph isprint isupper
0x5A isalnum isalpha isgraph isprint isupper
0x5B isgraph isprint ispunct
0x5C isgraph isprint ispunct
0x5D isgraph isprint ispunct
0x5E isgraph isprint ispunct
0x5F isgraph isprint ispunct
0x60 isgraph isprint ispunct
0x61 isalnum isalpha isgraph islower isprint isxdigit
0x62 isalnum isalpha isgraph islower isprint isxdigit
0x63 isalnum isalpha isgraph islower isprint isxdigit
0x64 isalnum isalpha isgraph islower isprint isxdigit
0x65 isalnum isalpha isgraph islower isprint isxdigit
0x66 isalnum isalpha isgraph islower isprint isxdigit
0x67 isalnum isalpha isgraph islower isprint
0x68 isalnum isalpha isgraph islower isprint
0x69 isalnum isalpha isgraph islower isprint
0x6A isalnum isalpha isgraph islower isprint
0x6B isalnum isalpha isgraph islower isprint
0x6C isalnum isalpha isgraph islower isprint
0x6D isalnum isalpha isgraph islower isprint
0x6E isalnum isalpha isgraph islower isprint
0x6F isalnum isalpha isgraph islower isprint
0x70 isalnum isalpha isgraph islower isprint
0x71 isalnum isalpha isgraph islower isprint
0x72 isalnum isalpha isgraph islower isprint
0x73 isalnum isalpha isgraph islower isprint
0x74 isalnum isalpha isgraph islower isprint
0x75 isalnum isalpha isgraph islower isprint
0x76 isalnum isalpha isgraph islower isprint
0x77 isalnum isalpha isgraph islower isprint
0x78 isalnum isalpha isgraph islower isprint
0x79 isalnum isalpha isgraph islower isprint
0x7A isalnum isalpha isgraph islower isprint
0x7B isgraph isprint ispunct
0x7C isgraph isprint ispunct
0x7D isgraph isprint ispunct
0x7E isgraph isprint ispunct
0x7F iscntrl
0x80
0x81
0x82
0x83
0x84
0x85
0x86
0x87
0x88
0x89
0x8A
0x8B
0x8C
0x8D
0x8E
0x8F
0x90
0x91
0x92
0x93
0x94
0x95
0x96
0x97
0x98
0x99
0x9A
0x9B
0x9C
0x9D
0x9E
0x9F
0xA0
0xA1
0xA2
0xA3
0xA4
0xA5
0xA6
0xA7
0xA8
0xA9
0xAA
0xAB
0xAC
0xAD
0xAE
0xAF
0xB0
0xB1
0xB2
0xB3
0xB4
0xB5
0xB6
0xB7
0xB8
0xB9
0xBA
0xBB
0xBC
0xBD
0xBE
0xBF
0xC0
0xC1
0xC2
0xC3
0xC4
0xC5
0xC6
0xC7
0xC8
0xC9
0xCA
0xCB
0xCC
0xCD
0xCE
0xCF
0xD0
0xD1
0xD2
0xD3
0xD4
0xD5
0xD6
0xD7
0xD8
0xD9
0xDA
0xDB
0xDC
0xDD
0xDE
0xDF
0xE0
0xE1
0xE2
0xE3
0xE4
0xE5
0xE6
0xE7
0xE8
0xE9
0xEA
0xEB
0xEC
0xED
0xEE
0xEF
0xF0
0xF1
0xF2
0xF3
0xF4
0xF5
0xF6
0xF7
0xF8
0xF9
0xFA
0xFB
0xFC
0xFD
0xFE
0xFF
